polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -1404,10 +1502,18 @@ fn map_candidates_to_cores<T: configuration::Config + scheduler::Config + inclus
 						);
 						break;
 					}
-					let maybe_injected_core_index: Option<CoreIndex> =
-						get_injected_core_index::<T>(allowed_relay_parents, &candidate);
 
-					if let Some(core_index) = maybe_injected_core_index {
+					// If candidate is v2, use the descriptor core index. We've already checked the
+					// core index is valid in `sanitize_backed_candidate_v2`. If candidate is v1,
+					// expect an injected core index.
+					let maybe_core_index =
```

Comment:
since we're adding this, we should probably remove the check on line 1490 for the ElasticScalingMVP feature. This should work without it being enabled